iceoryx_examples/ara_com/ara/com/method_server.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_ARA_COM_METHOD_SERVER_HPP
+#define IOX_EXAMPLES_ARA_COM_METHOD_SERVER_HPP
+
+#include "iceoryx_hoofs/cxx/optional.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
+#include "iceoryx_posh/popo/server.hpp"
+
+#include "ara/types.hpp"
+
+#include <memory>
+#include <utility>
+
+namespace ara
+{
+namespace com
+{
+class MethodServer
+{
+  public:
+    MethodServer(const core::String& service, const core::String& instance, const core::String& event) noexcept
+        : m_server({service, instance, event})
+    {
+        m_listener
+            .attachEvent(m_server,
+                         iox::popo::ServerEvent::REQUEST_RECEIVED,
+                         iox::popo::createNotificationCallback(onRequestReceived, *this))
+            .or_else([](auto) {
+                std::cerr << "unable to attach server" << std::endl;
+                std::exit(EXIT_FAILURE);
+            });
+    }
+
+    ~MethodServer() noexcept
+    {
+        m_listener.detachEvent(m_server, iox::popo::ServerEvent::REQUEST_RECEIVED);
+    }
+
+    ara::com::Future<AddResponse> computeSum(uint64_t addend1, uint64_t addend2)
```

Comment:
I do not understand why this is public. We should distinguish between immediate (synchronous) computation and deferred (async) computation.  The latter needs the future, the former does not.

Here we do not really utilize deferred computation, so why return a future?

There are three kinds of computation: 
1) immediate, same process - no futures required
2) deferred, local in the same process - needs futures
3) deferred, requested from other client - needs no future but uses request response and computation with 1)

We could have
4) deferred, requested from other client and computed async in the server process as well, but is this useful (it is costly)

Reply:
It's the API :grin: It is public to feel like a method call see `iox_ara_proxy.cpp`:

```
auto future = proxy.computeSum(addend1, addend2);
```